File: app/src/main/java/com/jcjm/desafioandroid/di/AppComponent.java
package com.jcjm.desafioandroid.di;


import com.jcjm.desafioandroid.MainActivity;
import com.jcjm.desafioandroid.activities.PullRequestsActivity;

import javax.inject.Singleton;

import dagger.Component;


@Component(modules = {UtilsModule.class})
@Singleton
public interface AppComponent {

    void doInjection(MainActivity mainActivity);

    void doInjection(PullRequestsActivity pullRequestsActivity);

}

File: app/src/main/java/com/anderson/apigithub_mvvm/di/AppComponent.kt
package com.anderson.apigithub_mvvm.di

import android.app.Application
import com.anderson.apigithub_mvvm.AppApplication
import com.anderson.apigithub_mvvm.di.builder.ActivityBuilder
import com.anderson.apigithub_mvvm.di.builder.ViewModelBuilder
import dagger.BindsInstance
import dagger.Component
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

/**
 * Created by anderson on 21/09/19.
 */
@Singleton
@Component(
    modules = [
        AppModule::class,
        AndroidSupportInjectionModule::class,
        ActivityBuilder::class,
        ViewModelBuilder::class
    ]
)
interface AppComponent {

    fun inject(application: AppApplication)

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): Builder

        fun build(): AppComponent
    }
}